src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ interface Item {
   [key: string]: any;
 }
 
-export function groupByKey(items, field) {
-  // write code here;
+export function groupByKey(items: Item[], field: string): object {
+  const finalArray: Item = {};
+
+  items.forEach((x: Item) => {
+    if (!finalArray[x[field]]) {
+      finalArray[x[field]] = [];
+      finalArray[x[field]].push(x);
+    } else {
+      finalArray[x[field]].push(x);
+    }
```

Comment:
don't repeat yourself
```suggestion
    if (!finalArray[x[field]]) {
      finalArray[x[field]] = [];
    }
    
    finalArray[x[field]].push(x);
```